test/ui/pagination_ui_tests.js

```diff
@@ -1,9 +1,89 @@
-var path = require('path')
-var renderTemplate = require(path.join(__dirname, '/../test_helpers/html_assertions.js')).render
+'use strict'
+
+// Local dependencies
+const {render} = require('../test_helpers/html_assertions.js')
 
 describe('The pagination links', function () {
-  it('should display correct pagination links', function () {
-    var templateData = {
+  it('should display correct pagination links for all filters', () => {
+    const filters = {
+      'reference': 'ref1',
+      'email': 'abc@def.com',
+      'state': 'Cancelled',
+      'selectedStates': [
+        'Cancelled'
+      ],
+      'payment_states': [
+        'cancelled'
+      ],
+      'fromDate': '2015-01-11',
+      'toDate': '2016-01-11',
+      'fromTime': '01:01:01',
+      'toTime': '02:02:02',
+      'pageSize': '100'
+    }
+
+    const templateData = transactionsTemplateData(filters)
+    const body = render('transactions/paginator', templateData)
+
+    body.should.containSelector('div.pagination')
+
+    templateData.paginationLinks.forEach((link) => {
+      body.should.containSelector('.paginationForm.page-' + link.pageName)
+      body.should.containSelector('.paginationForm.page-' + link.pageName + '  [name="state"]')
+        .withAttribute('value', 'Cancelled')
+      body.should.containSelector('.paginationForm.page-' + link.pageName + '  [name="reference"]')
+        .withAttribute('value', 'ref1')
+      body.should.containSelector('.paginationForm.page-' + link.pageName + ' [name="email"]')
+        .withAttribute('value', 'abc@def.com')
```

Comment:
Usually we use `example.com` or any other "invalid" domain name to make sure we don't use any real emails